samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoAvailableNotificationServiceImpl.java

```diff
@@ -0,0 +1,369 @@
+package org.sakaiproject.samigo.impl;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.sakaiproject.api.app.scheduler.ScheduledInvocationManager;
+import org.sakaiproject.authz.api.Member;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.email.api.EmailService;
+import org.sakaiproject.entity.api.ResourceProperties;
+import org.sakaiproject.event.api.NotificationService;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.samigo.api.SamigoAvailableNotificationService;
+import org.sakaiproject.samigo.util.SamigoConstants;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.time.api.UserTimeService;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.assessment.data.dao.assessment.ExtendedTime;
+import org.sakaiproject.tool.assessment.data.ifc.assessment.AssessmentBaseIfc;
+import org.sakaiproject.tool.assessment.facade.PublishedAssessmentFacade;
+import org.sakaiproject.tool.assessment.services.PersistenceService;
+import org.sakaiproject.tool.assessment.services.assessment.PublishedAssessmentService;
+import org.sakaiproject.tool.cover.ToolManager;
+import org.sakaiproject.user.api.Preferences;
+import org.sakaiproject.user.api.PreferencesService;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotDefinedException;
+import org.sakaiproject.util.ResourceLoader;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+@Slf4j
+public class SamigoAvailableNotificationServiceImpl implements SamigoAvailableNotificationService {
+
+    @Setter private EmailService emailService;
+    @Setter private PreferencesService preferencesService;
+    @Setter private ScheduledInvocationManager scheduledInvocationManager;
+    @Setter private ServerConfigurationService serverConfigurationService;
+    @Setter private SessionManager sessionManager;
+    @Setter private SiteService siteService;
+    @Setter private UserDirectoryService userDirectoryService;
+    @Setter private UserTimeService userTimeService;
+    private PublishedAssessmentService publishedAssessmentService;
+    private static final ResourceLoader rl = new ResourceLoader("SamigoAvailableNotificationMessages");
+    private final List<String> additionalHeaders = new ArrayList<>();
+    //feedback
+    public static Integer NO_FEEDBACK = Integer.valueOf(3);
+    //scoring types
+    public static final Integer HIGHEST_SCORE = 1;
+    public static final Integer LAST_SCORE= 2;
+    public static final Integer ALL_SCORE= 3;
+    public static final Integer AVERAGE_SCORE= 4;
+    //this matches DEAD_STATUS from AssessmentBaseIfc, indicating a deleted assessment
+    public static final Integer DEAD_STATUS = 2;
```

Comment:
These are already defined in `EvaluationModelIfc`. Same as above, rather than defining it here again, just reference the existing public fields like other files do.